pkgs/applications/networking/instant-messengers/gtkcord/default.nix

```diff
@@ -0,0 +1,46 @@
+{ makeDesktopItem, lib, buildGoModule, fetchFromGitHub, pkg-config, glib, go, gtk3, libhandy }:
+
+buildGoModule rec {
+  pname = "gtkcord";
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "diamondburned";
+    repo = "gtkcord3";
+    rev = "v${version}";
+    sha256 = "0n9600s50zpx8kv89icrlxarm3633gnizg1sq1zpask8bbv2xnfd";
+  };
+
+  nativeBuildInputs = [ pkg-config go ];
+  
+  buildInputs = [ glib gtk3 libhandy ];
+
+  desktopFile = makeDesktopItem {
+	  name = pname;
+    desktopName = "gtkcord";
+	  exec = "gtkcord3";
+	  icon = "gtkcord3";
+	  categories = "GTK;GNOME;Chat;";
+  };
+
+	preFixup = ''
+    # Create directory
+    install -d $out/share/applications/
```

Comment:
This is redundant.

Reply:
if I comment out that line I get ```install: target '/nix/store/bwq8pglciwwqmyq3d0cyh1c7hsi7ni4s-gtkcord-0.0.4/share/applications/' is not a directory: No such file or directory```

Reply:
Sorry, would likely need lowercase `-d` wen installing the desktop file.

Reply:
the `-d` option only exists to create directories. Not create and write. From the man page `install [OPTION]... -d DIRECTORY...`. Doing `install -d "${desktopFile}"/share/applications/* $out/share/applications/` does not work

Reply:
Right, sorry. For some reason the manpages I found on my phone are outdated so had to work from memory. On my computer, I have: 

> `-D` – create all leading components of `DEST` except the last, or all components of `--target-directory`, then copy `SOURCE` to `DEST`

so the proper invocation would be

```
    install -Dt $out/share/applications "${desktopFile}"/share/applications/*
```

Also you seem to have tab characters mixed in the indentation.

Reply:
Ok, fixed both issues